freqtrade/main.py

```diff
@@ -307,6 +307,30 @@ def min_roi_reached(trade: Trade, current_rate: float, current_time: datetime) -
     return False
 
 
+def should_sell(trade: Trade, rate: float, date: datetime, buy: bool, sell: bool) -> bool:
+    """
+    This function evaluate if on the condition required to trigger a sell has been reached
+    if the threshold is reached and updates the trade record.
+    :return: True if trade should be sold, False otherwise
+    """
+    # Check if minimal roi has been reached and no longer in buy conditions (avoiding a fee)
+    if min_roi_reached(trade, rate, date):
+        logger.debug('Executing sell due to ROI ...')
+        return True
+
+    # Experimental: Check if the trade is profitable before selling it (avoid selling at loss)
+    if _CONF.get('experimental', {}).get('sell_profit_only', False):
+        logger.debug('Checking if trade is profitable ...')
+        if not buy and trade.calc_profit(rate=rate) <= 0:
```

Comment:
I think this "not buy" condition makes no sense here? (I know it's not from your PR but we might cleanup the logic here). If the buy signal is true, then we should skip the profit check? That's not correct.

Reply:
I like this comment, let’s target it in another PR with test for it. It will be easier to review.

Reply:
I should read the whole PR changes before commenting.